File: src/micapolos/tata8/Composite.java
package micapolos.tata8;

import micapolos.awt.MultiplyComposite;
import micapolos.awt.SoftLightComposite;

import java.awt.*;

public enum Composite {
  NORMAL(AlphaComposite.SrcOver),
  SOFT_LIGHT(SoftLightComposite.INSTANCE),
  MULTIPLY(MultiplyComposite.INSTANCE);

  final java.awt.Composite awt;

  Composite(java.awt.Composite awt) {
    this.awt = awt;
  }
}

File: src/micapolos/zexy/compiler/Composite.kt
package micapolos.zexy.compiler

import micapolos.zexy.indexed.Composite
import micapolos.tata8.Composite as TataComposite

val Composite.tata8 get() =
  when (this) {
    Composite.NORMAL -> TataComposite.NORMAL
    Composite.SOFT_LIGHT -> TataComposite.SOFT_LIGHT
    Composite.MULTIPLY -> TataComposite.MULTIPLY
  }